### PR TITLE
#SB-15205 | L1 Version of MHRD, PMO, CM Dashboards - ECG of Learning

### PR DESCRIPTION
**Pull request by kumarks1122**
This shows TPS across a week. It is updated on an hourly basis. This graph will be configured in the dashboard page of MHRD. The data for the Graph will be taken from Prometheus DB by using API.

### CODE REVIEW

File: scripts/ecg_learning.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    schema = StructType([
+        StructField('time', IntegerType(), True),
+        StructField('tps', StringType(), True)
+    ])
+
+    tps_df = spark.createDataFrame(ecg_data_rdd,schema)
+
+    tps_df = tps_df.withColumn("tps", tps_df["tps"].cast("float"))
+    tps_df = tps_df.withColumn("tps", F.ceil(tps_df["tps"]))
+    tps_df = tps_df.withColumn("time", F.from_unixtime(tps_df["time"], "yyyy/MM/dd hh:mm:ss"))
```
Comment (SanthoshVasabhaktula) [inline scripts/ecg_learning.py]: Do we need to add timezone information here? if it is running in a server configured on UTC, will the time be shown in UTC?

File: scripts/ecg_learning.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    response = requests.request("GET", url, headers=headers, params=querystring)
+
+    return response.json()["data"]["result"][0]["values"]
+
+
+def remove_last_day(df):
+    current_hour = current_time.hour
+    if current_hour == 0 & df.count() >= 672:
```
Comment (SanthoshVasabhaktula) [inline scripts/ecg_learning.py]: I am assuming that this only deletes the first day data only after the current day is appended.

File: scripts/ecg_learning.py (hunk 1 above)
Comment (kumarks1122) [reply scripts/ecg_learning.py]: I used the server's timezone. In the new commit, I have updated the script for the Indian timezone also.

File: scripts/ecg_learning.py (hunk 2 above)
Comment (kumarks1122) [reply scripts/ecg_learning.py]: I was deleting the first day's data before merging the old and 1-hour dataframe. In the new commit, I have updated the script for deleting the first-day data only after the current day is appended.

**Pull request merged**